src/promptflow-evals/promptflow/evals/synthetic/xpia_simulator.py

```diff
@@ -0,0 +1,214 @@
+         - '**$schema**': A string indicating the schema URL for the conversation format.
+
+         The 'content' for 'assistant' role messages may includes the messages that your callback returned.
+        :rtype: Dict[str, [List[Dict[str, Any]]]] with two elements
+
+        **Output format**
+
+        .. code-block:: python
+
+            return_value = {
+                "jailbreak": [
+                {
+                    'template_parameters': {},
+                    'messages': [
+                        {
+                            'content': '<jailbreak prompt> <adversarial question>',
+                            'role': 'user'
+                        },
+                        {
+                            'content': "<response from endpoint>",
+                            'role': 'assistant',
+                            'context': None
+                        }
+                    ],
+                    '$schema': 'http://azureml/sdk-2-0/ChatConversation.json'
+                }],
+                "regular": [
+                {
+                    'template_parameters': {},
+                    'messages': [
+                    {
+                        'content': '<adversaril question>',
+                        'role': 'user'
+                    },
+                    {
+                        'content': "<response from endpoint>",
+                        'role': 'assistant',
+                        'context': None
+                    }],
+                    '$schema': 'http://azureml/sdk-2-0/ChatConversation.json'
+                }]
+            }
+        """
+        if scenario not in AdversarialScenario.__members__.values():
+            raise ValueError("Invalid adversarial scenario")
+        regular_sim = AdversarialSimulator(azure_ai_project=self.azure_ai_project, credential=self.credential)
+        regular_sim_results = await regular_sim(
+            scenario=scenario,
+            target=target,
+            max_conversation_turns=max_conversation_turns,
+            max_simulation_results=max_simulation_results,
+            api_call_retry_limit=api_call_retry_limit,
+            api_call_retry_sleep_sec=api_call_retry_sleep_sec,
+            api_call_delay_sec=api_call_delay_sec,
+            concurrent_async_task=concurrent_async_task,
+            jailbreak=False,
+        )
+        jb_sim = AdversarialSimulator(azure_ai_project=self.azure_ai_project, credential=self.credential)
+        jb_sim_results = await jb_sim(
+            scenario=scenario,
+            target=target,
+            max_conversation_turns=max_conversation_turns,
+            max_simulation_results=max_simulation_results,
+            api_call_retry_limit=api_call_retry_limit,
+            api_call_retry_sleep_sec=api_call_retry_sleep_sec,
+            api_call_delay_sec=api_call_delay_sec,
+            concurrent_async_task=concurrent_async_task,
+            jailbreak=True,
```

Comment:
Since this is a publicly documented kwarg, it is confusing to have one be just `jailbreak` and then another acronym be `xpia` when `xpia` *is a type of jailbreak*. I would prefer something like `direct_jailbreak`(currently jailbreak) and `indirect_jailbreak` (currently xpia) to disambiguate + maintain consistency.

Reply:
+1. There shouldn't be 2 ways to do the same thing. With dedicated direct/Indirect attack simulator classes, we should consider hiding `jailbreak` and `xpia` parameters.

Another possibility is to get rid of dedicated classes for direct/indirect attack simulators. Instead, provide helper functions to generate xpia/upia datasets.

E.g.

```python

from promptflow.evals.synthetic import generate_xpia_dataset, generate_upia_dataset

adv_sim = AdversarialSimulator()

xpia_conv = generate_xpia_dataset(adv_sim, ...)

upia_conv = generate_upia_dataset(adv_sim, ...)

```

Reply:
Agree with suggestions above. Should name also match with evaluator names (direct attack and indirect attack)

Reply:
Discussed offline - will table helper function proposals for now but params have been updated